docs/developer-guide/partition-data.rst

```diff
@@ -10,26 +10,69 @@ of the join have the same configuration settings for partitions.
 Co-partitioning Requirements
 ****************************
 
-* The input records for the join must have the :ref:`same keying scheme <keys-have-same-keying-scheme>`
+* The input records for the join must have the :ref:`same key schema <keys-have-same-key-schema>`
 * The input records must have the :ref:`same number of partitions <keys-have-same-number-of-partitions>` on both sides.
 * Both sides of the join must have the :ref:`same partitioning strategy <records-have-same-partitioning-strategy>`.  
 
 When your inputs are co-partitioned, records with the same key, from both
 sides of the join, are delivered to the same stream task during processing.
-If your inputs aren't co-partitioned, you need to :ref:`re-key one of the
-them <ensure-co-partitioning>` by using the PARTITION BY clause.
 
-.. _keys-have-same-keying-scheme:
+.. _keys-have-same-key-schema:
 
-Records Have the Same Keying Scheme
-===================================
+Records have the Same Key Schema
+================================
 
-The input records for the join must have the same keying scheme, which means
-that the join must use the same key field on both sides.
+For a join to work the keys from both sides must have the same serialized binary data.
 
-For example, you can join a stream of user clicks that's keyed by a ``VARCHAR userId``
-field with a table of user profiles that's keyed by a ``VARCHAR userId`` field. 
-The join won't match if the key fields don't have the same name and type.
+For example, you can join a stream of user clicks that's keyed on a ``VARCHAR`` user id with a table
+of user's profiles that's also keyed on a ``VARCHAR`` user id. Records with the exact same user id
+on both sides will be joined.
+
+Tables created on top of existing Kafka topics, for example those created with a ``CREATE TABLE``
+statement, will be keyed on the data held in the key of the records in the
+Kafka topic.  KSQL presents this data in the ``ROWKEY`` column, and expects the data to be a ``VARCHAR``.
+
+Tables created inside KSQL from other sources, for example those created with a ``CREATE TABLE AS SELECT``
+statement, will copy the key from their source(s) unless there is an explicit ``GROUP BY`` or ``PARTITION BY``
+clause, which can changed what the table is keyed on.
+
+KSQL will automatically repartition a stream of data before joining, if required. In the example below
+KSQL is used to join a stream of clicks with a table of users.  The ``clicks`` stream is not keyed on anything
+initially. The ``users`` table keyed on ``userId``.  Because the stream is being joined on a column other than
+it's key field, i.e. ``ROWKEY``, KSQL will automatically repartition the stream under the hood.

```

Comment:
The writing is a bit confusing: you say the stream has no key but then you say the "the stream is joined on a column other than it's key field". I thought it doesn't have a key?

Reply:
Slight update applied.